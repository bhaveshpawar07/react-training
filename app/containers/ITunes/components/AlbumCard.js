import React from 'react';
import PropTypes from 'prop-types';
import T from '@components/T';
import If from '@components/If';
import { Col } from 'antd';
import styled from 'styled-components';
const CustomCard = styled(Col)`
  && {
    width: 150px;
    display: flex;
    flex-direction: column;
  }
`;
const CustomImg = styled.img`
  && {
    border-radius: ${(props) => props.borderRadius}px;
  }
`;
const CustomDiv = styled.div`
  && {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 10px;
  }
`;
const CustomT = styled(T)`
  && {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    font-weight: ${(props) => props.fontWeight};
    font-size: ${(props) => props.fontSize}px;
  }
`;
export const AlbumCard = ({ artworkUrl100, trackName, trackExplicitness, artistName }) => {
  return (
    <CustomCard span={6}>
      <CustomImg src={artworkUrl100} borderRadius={20} />
      <CustomDiv>
        <CustomT text={trackName} fontWeight={'bold'} />
        <If condition={trackExplicitness === 'explicit'}>
          <CustomImg
            src="https://cdn-icons.flaticon.com/png/512/3097/premium/3097003.png?token=exp=1661161542~hmac=99d22989a8dc8287886e84079bddb458"
            width="15px"
          />
        </If>
      </CustomDiv>
      <CustomT text={artistName} fontSize={11} />
    </CustomCard>
  );
};

AlbumCard.propTypes = {
  artworkUrl100: PropTypes.string,
  trackName: PropTypes.string,
  trackExplicitness: PropTypes.string,
  artistName: PropTypes.string
};

export default AlbumCard;
