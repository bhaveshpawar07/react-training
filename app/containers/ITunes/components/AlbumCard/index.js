import React from 'react';
import PropTypes from 'prop-types';
import T from '@components/T';
import If from '@components/If';
import { Col } from 'antd';
import styled from 'styled-components';
const CustomCard = styled(Col)`
  && {
    width: 150px;
    display: flex;
    flex-direction: column;
  }
`;
const CustomImg = styled.img`
  && {
    border-radius: ${(props) => props.borderRadius}px;
  }
`;
const CustomDiv = styled.div`
  && {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 10px;
  }
`;
const CustomT = styled(T)`
  && {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    font-weight: ${(props) => props.fontWeight};
    font-size: ${(props) => props.fontSize}px;
  }
`;
export const AlbumCard = ({ artworkUrl100, trackName, trackExplicitness, artistName, index, musicPlayer }) => {
  return (
    <CustomCard span={6} data-testid="album-card">
      <CustomImg src={artworkUrl100} borderRadius={20} onClick={() => musicPlayer(index)} />
      <CustomDiv>
        <CustomT data-testid="track-name" text={trackName} fontWeight={'bold'} />
        <If condition={trackExplicitness === 'explicit'}>
          <CustomImg
            src="https://cdn-icons.flaticon.com/png/512/3097/premium/3097003.png?token=exp=1661161542~hmac=99d22989a8dc8287886e84079bddb458"
            width="15px"
          />
        </If>
      </CustomDiv>
      <CustomT data-testid="artist-name" text={artistName} fontSize={11} />
    </CustomCard>
  );
};

AlbumCard.propTypes = {
  artworkUrl100: PropTypes.string,
  trackName: PropTypes.string,
  trackExplicitness: PropTypes.string,
  artistName: PropTypes.string,
  index: PropTypes.number,
  musicPlayer: PropTypes.func
};

export default AlbumCard;
