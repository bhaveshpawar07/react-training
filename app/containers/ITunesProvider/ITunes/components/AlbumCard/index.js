import React from 'react';
import PropTypes from 'prop-types';
import T from '@components/T';
import If from '@components/If';
import { Col } from 'antd';
import styled from 'styled-components';
import { useHistory } from 'react-router-dom';
const CustomCard = styled(Col)`
  && {
    width: 150px;
    display: flex;
    flex-direction: column;
  }
`;
const CustomImg = styled.img`
  && {
    border-radius: ${(props) => props.borderRadius}px;
  }
`;
const CustomDiv = styled.div`
  && {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-top: 10px;
  }
`;
const CustomT = styled(T)`
  && {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    font-weight: ${(props) => props.fontWeight};
    font-size: ${(props) => props.fontSize}px;
  }
`;

export const AlbumCard = ({ trackId, artworkUrl100, trackName, trackExplicitness, artistName, index, musicPlayer }) => {
  const history = useHistory();
  const showDetails = (trackId) => {
    history.push(`/details/${trackId}`);
  };
  return (
    <CustomCard span={6} data-testid="album-card">
      <CustomImg
        src={artworkUrl100}
        borderRadius={20}
        data-testid="album-songImage"
        onClick={() => musicPlayer(index)}
      />
      <CustomDiv data-testid="album-viewDetails" onClick={() => showDetails(trackId)}>
        <CustomT data-testid="track-name" text={trackName} fontWeight={'bold'} />
        <If condition={trackExplicitness === 'explicit'}>
          <CustomImg
            src="https://cdn-icons.flaticon.com/png/512/3097/premium/3097003.png?token=exp=1661161542~hmac=99d22989a8dc8287886e84079bddb458"
            width="15px"
          />
        </If>
      </CustomDiv>
      <CustomT data-testid="artist-name" text={artistName} fontSize={11} />
    </CustomCard>
  );
};

AlbumCard.propTypes = {
  trackId: PropTypes.number,
  artworkUrl100: PropTypes.string,
  trackName: PropTypes.string,
  trackExplicitness: PropTypes.string,
  artistName: PropTypes.string,
  index: PropTypes.number,
  musicPlayer: PropTypes.func
};

export default AlbumCard;
