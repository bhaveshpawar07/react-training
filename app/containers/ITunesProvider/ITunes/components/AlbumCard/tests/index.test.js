import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { renderProvider, renderWithIntl, timeout } from '@utils/testUtils';
import AlbumCard from '../index';
import { useHistory } from 'react-router-dom';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
describe('<AlbumCard /> tests', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<AlbumCard />);
    expect(baseElement).toMatchSnapshot();
  });
  it('should contain 1 album card component', () => {
    const { getAllByTestId } = renderWithIntl(<AlbumCard />);
    expect(getAllByTestId('album-card').length).toBe(1);
  });
  it('should render album details inside the card', () => {
    const trackName = 'test';
    const artistName = 'testArtist';
    const { getByTestId } = renderWithIntl(<AlbumCard trackName={trackName} artistName={artistName} />);
    expect(getByTestId('track-name')).toHaveTextContent(trackName);
    expect(getByTestId('artist-name')).toHaveTextContent(artistName);
  });
  it('should render explicit svg if the track is explicit', () => {
    const trackName = 'test';
    const trackExplicitness = 'explicit';
    const { getByTestId } = renderWithIntl(<AlbumCard trackName={trackName} trackExplicitness={trackExplicitness} />);
    expect(getByTestId('track-name')).toHaveTextContent(trackName);
    expect(getByTestId('explicit-svg')).toBeInTheDocument;
  });
  it('should call musicPlayer with proper index after a album is clicked', () => {
    const index = 1;
    const musicPlayer = jest.fn();
    const { getByTestId } = renderProvider(<AlbumCard index={index} musicPlayer={musicPlayer} />);
    fireEvent.click(getByTestId('album-songImage'));
    expect(musicPlayer).toBeCalledWith(index);
  });
  it('should redirect to AlbumDetails when clicked on clickable component', async () => {
    const history = createBrowserHistory();
    const trackId = 1;

    const { getByTestId } = renderProvider(
      <Router history={history}>
        <AlbumCard trackId={trackId} />
      </Router>
    );
    const h = useHistory();
    const historySpy = jest.spyOn(h, 'push');
    fireEvent.click(getByTestId('album-viewDetails'));
    await timeout(500);
    expect(historySpy).toHaveBeenCalledWith(`/details/${trackId}`);
  });
});
