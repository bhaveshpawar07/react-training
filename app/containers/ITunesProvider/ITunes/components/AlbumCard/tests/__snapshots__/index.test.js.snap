// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumCard /> tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-col ant-col-6 AlbumCard__CustomCard-sc-1u39i60-0 nKPfE"
      data-testid="album-card"
    >
      <img
        class="AlbumCard__CustomImg-sc-1u39i60-1 cRwCbw"
        data-testid="album-songImage"
      />
      <div
        class="AlbumCard__CustomDiv-sc-1u39i60-2 fnzPve"
        data-testid="album-viewDetails"
      >
        <p
          class="T__StyledText-gjlic1-0 kYcyRq AlbumCard__CustomT-sc-1u39i60-3 eERINR"
          data-testid="track-name"
          font-weight="bold"
        />
      </div>
      <p
        class="T__StyledText-gjlic1-0 kYcyRq AlbumCard__CustomT-sc-1u39i60-3 iMnshM"
        data-testid="artist-name"
        font-size="11"
      />
    </div>
  </div>
</body>
`;
